cc/gtp_player.cc

```diff
@@ -115,13 +116,55 @@ void GtpPlayer::Run() {
   stdin_thread.join();
 }
 
+void GtpPlayer::NewGame() {
+  node_to_info_.clear();
+  id_to_info_.clear();
+  to_eval_.clear();
+  MctsPlayer::NewGame();
+  RegisterNode(root());
+}
+
 Coord GtpPlayer::SuggestMove() {
   if (courtesy_pass_ && root()->move == Coord::kPass) {
     return Coord::kPass;
   }
   return MctsPlayer::SuggestMove();
 }
 
+bool GtpPlayer::PlayMove(Coord c) {
```

Comment:
can you comment a bit about how this is different than genmove / make explicit the linkages here w.r.t. the front end?

Reply:
I've added a TODO to gtp_player.h to document all the Handle* methods of GtpPlayer. I'll do a pass over the comments & documentation once the important features are mostly finished